src/main/kotlin/org/pkl/lsp/ast/Member.kt

```diff
@@ -51,7 +51,34 @@ class PklClassPropertyImpl(
 
   override val type: PklType? by lazy { typeAnnotation?.type }
 
-  override val isDefinition: Boolean by lazy { expr != null }
+  override fun isDefinition(context: PklProject?): Boolean =
+    when {
+      type != null -> true
+      isLocalOrConstOrFixed -> true
+      else -> {
+        when (val owner = this.parentOfTypes(PklModule::class, PklClass::class)) {
+          is PklModule -> {
+            if (owner.header?.moduleExtendsAmendsClause?.isAmend == true) false
+            else
+              when (val supermodule = owner.supermodule(context)) {
+                null ->
+                  !project.pklBaseModule.moduleType.ctx.cache(context).properties.containsKey(name)
+                else -> !supermodule.cache(context).properties.containsKey(name)
+              }
+          }
+          is PklClass -> {
+            owner.superclass(context)?.let { superclass ->
+              return !superclass.cache(context).properties.containsKey(name)
+            }
+            owner.supermodule(context)?.let { supermodule ->
+              return !supermodule.cache(context).properties.containsKey(name)
+            }
+            true
+          }
+          else -> false
+        }
+      }
+    }
```

Comment:
This implementation is copied from https://github.com/apple/pkl-intellij/blob/255491bc6ced4d653bf31fa17e8935203db7e3f5/src/main/kotlin/org/pkl/intellij/psi/PklClassPropertyBase.kt#L52